### PR TITLE
Add types for error object

### LINKED ISSUE #1114: `ErrorObject` attributes are not visible to mypy
### Describe the bug

Since type annotations have been exported, many errors are now being reported for missing attributes on `ErrorObject` instances.

### To Reproduce

in `t.py`:

```python
import stripe

def f() -> None:
    try:
        stripe.PaymentIntent.create()
    except stripe.error.InvalidRequestError as e:
        error = e.construct_error_object()
        assert error is not None  # not relevant to this issue
        print(error.code)
```

```console
$ mypy t.py
t.py:9: error: "ErrorObject" has no attribute "code"  [attr-defined]
Found 1 error in 1 file (checked 1 source file)
```

### Expected behavior

No type checking errors.

### Code snippets

_No response_

### OS

macOS

### Language version

observed on 3.10.11 and 3.11.5

### Library version

7.1.0

### API version

N/A

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by richardm-stripe**
Fixes #1114 

Sanity checked against our definitions in [another library](https://github.com/stripe/stripe-node/blob/b9e3b91a3f68e9ef7de0f6d0a21c7f94cb92ae9c/types/Errors.d.ts#L13)

### CODE REVIEW

**Review by anniel-stripe — APPROVED** [review]

File: stripe/api_resources/error_object.py [hunk 1]
```diff
@@ -1,8 +1,26 @@
+from typing import Optional
+from stripe.api_resources.payment_intent import PaymentIntent
+from stripe.api_resources.setup_intent import SetupIntent
+from stripe.api_resources.source import Source
 from stripe.util import merge_dicts
 from stripe.stripe_object import StripeObject
 
+from api_resources.payment_method import PaymentMethod
```
Comment (anniel-stripe) [inline stripe/api_resources/error_object.py:8]: This import looks different from `PaymentIntent`, etc. above?

**Review by anniel-stripe — COMMENTED** [review]
Oops approved too soon

File: stripe/api_resources/error_object.py (hunk 1 above)
Comment (richardm-stripe) [reply stripe/api_resources/error_object.py:8]: Oh, good catch. Autoimport

File: stripe/api_resources/error_object.py [hunk 2]
```diff
@@ -1,8 +1,27 @@
+from typing import Optional
+from typing_extensions import TYPE_CHECKING
 from stripe.util import merge_dicts
 from stripe.stripe_object import StripeObject
 
+if TYPE_CHECKING:
+    from stripe.api_resources.payment_intent import PaymentIntent
+    from stripe.api_resources.setup_intent import SetupIntent
+    from stripe.api_resources.source import Source
+    from stripe.api_resources.payment_method import PaymentMethod
```
Comment (richardm-stripe) [inline stripe/api_resources/error_object.py:10]:
```suggestion
    from stripe.api_resources.payment_method import PaymentMethod

```

**Pull request merged**